// @expo/configure-splash-screen/build/android/Colors.xml.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const color_string_1 = __importDefault(require("color-string"));
const path_1 = __importDefault(require("path"));
const xml_manipulation_1 = require("../xml-manipulation");
const COLORS_XML_FILE_PATH = './res/values/colors.xml';
const COLORS_NIGHT_XML_FILE_PATH = './res/values-night/colors.xml';
function ensureDesiredXmlContent(xml, { backgroundColor, statusBarBackgroundColor, }) {
    let idx = 0;
    const result = xml_manipulation_1.mergeXmlElements(xml, {
        elements: [
            {
                name: 'resources',
                elements: [
                    {
                        idx: idx++,
                        comment: ` Below line is handled by '@expo/configure-splash-screen' command and it's discouraged to modify it manually `,
                    },
                    {
                        deletionFlag: !backgroundColor,
                        idx: !backgroundColor ? undefined : idx++,
                        name: 'color',
                        attributes: {
                            name: 'splashscreen_background',
                        },
                        elements: [
                            {
                                text: backgroundColor ? getAndroidStyleHex(backgroundColor) : '',
                            },
                        ],
                    },
                    {
                        deletionFlag: !statusBarBackgroundColor,
                        idx: !statusBarBackgroundColor ? undefined : idx++,
                        name: 'color',
                        attributes: {
                            name: 'splashscreen_statusbar_color',
                        },
                        elements: [
                            {
                                text: statusBarBackgroundColor ? getAndroidStyleHex(statusBarBackgroundColor) : '',
                            },
                        ],
                    },
                ],
            },
        ],
    });
    return result;
}
/**
 * css-recognized hex is of format `#RRGGBB(AA)` or `#RGB(A)`, while Android accepts `#(AA)RRGGBB` or `#(A)RGB` (https://developer.android.com/guide/topics/resources/color-list-resource)
 * This function converts following formats:
 * - `#RRGGBBAA` ➡️ `#AARRGGBB`,
 * - `#RGBA` ➡️ `#ARGB`.
 */
function getAndroidStyleH
